cpp/sopt/CMakeLists.txt

```diff
@@ -68,6 +72,11 @@ if(SOPT_MPI)
   target_link_libraries(sopt ${MPI_CXX_LIBRARIES})
   target_include_directories(sopt SYSTEM PUBLIC ${MPI_CXX_INCLUDE_PATH})
 endif()
+if (cppflow)
```

Comment:
@tkoskela we wanted to use `if(TENSORFLOW_LIB_FOUND)` here rather than `if(cppflow)` but that variable isn't created by the `find_library` function, only by `find_package`. The [docs for find_library](https://cmake.org/cmake/help/latest/command/find_library.html) say

> If the library is found the result is stored in the variable and the search will not be repeated unless the variable is cleared. If nothing is found, the result will be <VAR>-NOTFOUND.

But I'm not sure how to check for that here. Would it be something like something like:
`if(TENSORFLOW_LIB != "TENSORFLOW_LIB-NOTFOUND"` ??

Reply:
Let's use `if(cppflow)` then.

The other would be to find it with `find_package()` instead, but it looks like we would then have to write the cmake package file for it.

Reply:
Ah, there might be a bigger problem with `find_library`. It only finds a library object, not the include directory.

Reply:
Actually we don't seem to need TensorFlow include dirs in the `target_include_directories` command, `find_library` seems to do the job.

Reply:
We have Tensorflow set as REQUIRED in the `dependencies.cmake` so I guess it will stop if Tensorflow is not found, so we don't need to worry about it trying to run this block when Tensorflow isn't installed.